### PR TITLE
Assertions cleanup

### PR DESCRIPTION
**Pull request by shindere**
Replace a few occurrences of assert and many occurrences of Assert
by CAMLassert.

For multi-line assertions, an effort has been made to keep the code
both aligned correctly and beyond the 80-characters-per-line limit.
Sighted input would be welcome to make sure this has actually been achieved.

Also, in byterun/caml/weak.h, function caml_ephe_clean, does
anybody know why the assertions have not been aligned with

  header_t hd; ?

Does this need to be preserved or changed?

Similar question for byterun/compact.c, byterun/minor_gc.c, byterun/misc.c and
byterun/freelist.c.

In this file, does the big comment with all the assertions in it
need to be preserved?

In byterun/intern.c, see the comment saying: "do not do the urgent_gc
check here because it might darken intern_block into gray and break
the Assert 3 lines down.": maybe "3 lines down" could be replaced by "below"?

### CODE REVIEW

**Comment by xavierleroy** [comment]
> does anybody know why the assertions have not been aligned ?  

Assertions hanging to the right of the page are one of @damiendoligez's mannerisms.

> Does this need to be preserved or changed?

Preserved, of course, like all mannerisms from great masters.

**Comment by gasche** [comment]
> in `byterun/intern.c` [...] maybe "3 lines down" could be replaced by "below"?

I wondered which of the two asserts below was being meant. The "3 lines" measure was broken by ddd51af088bed9c0df7e73da161d843666256e33, and the relevant assert is (as could be expected) the first one, that is the `intern_color` Assert. I would rephrase this as "the intern_color Assert below".

**Comment by shindere** [comment]
Gabriel Scherer (2017/03/10 10:35 -0800):
> > in `byterun/intern.c` [...] maybe "3 lines down" could be replaced by "below"?
> 
> I wondered which of the two asserts below was being meant. The "3
> lines" measure was broken by ddd51af088bed9c0df7e73da161d843666256e33,
> and the relevant assert is (as could be expected) the first one, that
> is the `intern_color` Assert. I would rephrase this as "the
> intern_color Assert below".

Done, thanks. It now says "and break the intern_color assertion below".

Xavier Leroy (2017/03/10 10:27 -0800):
> > does anybody know why the assertions have not been aligned ?  
> 
> Assertions hanging to the right of the page are one of
> @damiendoligez's
> mannerisms.
> 
> > Does this need to be preserved or changed?
> 
> Preserved, of course, like all mannerisms from great masters.

Okay :)

**Comment by shindere** [comment]
Just rebased on latest trunk. Does it look okay to be merged, now?

**Comment by damiendoligez** [comment]
> Preserved, of course, like all mannerisms from great masters.

Nah, I've grown out of this one...

**Comment by shindere** [comment]
Damien Doligez (2017/03/14 09:04 -0700):
> > Preserved, of course, like all mannerisms from great masters.
> 
> Nah, I've grown out of this one...

Pretty cool, cause (1) it's not so uniform accross OCaml's source
code and (2) it's not that convenient from an a11y (accessibility, like
i18n for internationalization) perspective: when going through the code
with down arrow and cursor at the beginning of lines, if an assertion
starts at character 41 or more, I'll see what looks like a blank line
and completely miss it. Unless, of course, I am in search for it, in
which case I'll scroll and see it.

**Comment by damiendoligez** [comment]
> (2) it's not that convenient from an a11y (accessibility, like
i18n for internationalization) perspective

This is indeed a very good reason to remove it.

**Comment by shindere** [comment]
Damien Doligez (2017/03/15 09:13 -0700):
> > (2) it's not that convenient from an a11y (accessibility, like
> i18n for internationalization) perspective
> 
> This is indeed a very good reason to remove it.

Done in commit 0aab402e739dad4c79a7c16b7bce38bed3588752 which has just
been pushed to trunk.
